torch_frame/datasets/amphibians.py

```diff
@@ -0,0 +1,102 @@
+import os.path as osp
+import zipfile
+
+import pandas as pd
+
+import torch_frame
+
+
+class Amphibians(torch_frame.data.Dataset):
+    r"""The Amphibians
+    <https://archive.ics.uci.edu/dataset/528/amphibians>`_
+    dataset. It's a task to predict which of the 7 frogs types appeared
+    in the habitat.
+
+    **STATS:**
+
+    .. list-table::
+        :widths: 10 10 10 10 20 10
+        :header-rows: 1
+
+        * - #rows
+          - #cols (numerical)
+          - #cols (categorical)
+          - #cols (text_embedded)
+          - Task
+          - Missing value ratio
+        * - 189
+          - 3
+          - 20
+          - 0
+          - multilabel classification
+          - 0.0%
+    """
+    url = 'https://archive.ics.uci.edu/static/public/528/amphibians.zip'
+
+    def __init__(self, root: str):
+        path = self.download_url(self.url, root)
+        folder_path = osp.dirname(path)
+
+        with zipfile.ZipFile(path, 'r') as zip_ref:
+            zip_ref.extractall(folder_path)
+
+        data_path = osp.join(folder_path, 'dataset.csv')
+        names = [
+            'ID',
+            'MV',
+            'SR',
+            'NR',
+            'TR',
+            'VR',
+            'SUR1',
+            'SUR2',
+            'SUR3',
+            'UR',
+            'FR',
+            'OR',
+            'RR',
+            'BR',
+            'MR',
+            'CR',
+            't1',
+            't2',
+            't3',
+            't4',
+            't5',
+            't6',
+            't7',
+        ]
+        df = pd.read_csv(data_path, names=names, sep=';')
+        # Drop the first 2 rows containing metadata
+        df = df.iloc[2:].reset_index(drop=True)
+        target_cols = ['t1', 't2', 't3', 't4', 't5', 't6', 't7']
+        df['t'] = df.apply(
+            lambda row: '|'.join(
+                [col for col in target_cols if row[col] == '1']), axis=1)
+        df = df.drop(target_cols, axis=1)
```

Comment:
Can we use the `list` instead, since that's more robust? We support list element in multi-categorical column.